packages/admin-ui/src/filters/filter-listbox.tsx

```diff
@@ -1,43 +1,48 @@
 import React, { useEffect, useRef } from 'react'
-
 import { ComboboxList } from 'ariakit/combobox'
-
-import { Box, tag } from '..'
+import { createComponent, useElement } from '@vtex/admin-ui-react'
 
 import * as style from './filter.style'
 import { usePopoverContext } from './filter-popover-context'
-
-export const FilterListbox = (props: FilterListboxProps) => {
-  const { children, id } = props
-  const optionsContainerRef = useRef<HTMLDivElement>(null)
-
-  const {
-    setIsScrollableLayout,
-    state: { combobox },
-  } = usePopoverContext()
-
-  const scrollHeight = optionsContainerRef?.current?.scrollHeight ?? 0
-  const containerHeight = optionsContainerRef?.current?.clientHeight ?? 0
-
-  useEffect(() => {
-    setIsScrollableLayout(scrollHeight > containerHeight)
-  }, [scrollHeight, containerHeight])
-
-  const ariakitcomboboxState = { ...combobox, matches: [] }
-
-  return (
-    <tag.span ref={optionsContainerRef} csx={style.scrollableContainer}>
-      <Box
-        as={ComboboxList as any}
-        state={{ ...ariakitcomboboxState, visible: true }}
-        id={id}
-        csx={style.list}
-      >
-        {children}
-      </Box>
-    </tag.span>
-  )
-}
+import { Box } from '..'
+
+export const FilterListbox = createComponent<'span', FilterListboxProps>(
+  (props) => {
+    const { children, id, ...restProps } = props
+    const optionsContainerRef = useRef<HTMLDivElement>(null)
+
+    const {
+      setIsScrollableLayout,
+      state: { combobox },
+    } = usePopoverContext()
+
+    const scrollHeight = optionsContainerRef?.current?.scrollHeight ?? 0
+    const containerHeight = optionsContainerRef?.current?.clientHeight ?? 0
+
+    useEffect(() => {
+      setIsScrollableLayout(scrollHeight > containerHeight)
+    }, [scrollHeight, containerHeight])
+
+    const ariakitcomboboxState = { ...combobox, matches: [] }
+
+    return useElement('span', {
```

Comment:
I think this should be a `div` because it is a problem have a inline element `span` wrapping a block element `div`.